aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutboxTest.java

```diff
@@ -650,12 +655,40 @@ public void enqueueEventPostedWhenNewOutboxItemAdded() {
         mutationOutbox.enqueue(PendingMutation.deletion(BlogOwner.builder()
             .name("Tony Swanson")
             .build(), BlogOwner.class
-        )).blockingAwait();
+        )).blockingAwait(TIMEOUT_MS, TimeUnit.MILLISECONDS);
 
         // Assert: we got an event!
         eventsObserver.awaitCount(1)
             .assertNoErrors()
             .assertNotTerminated()
             .assertValue(OutboxEvent.CONTENT_AVAILABLE);
     }
+
+    /**
+     * If the queue contains multiple items, then
+     * {@link PersistentMutationOutbox#nextMutationForModelId(String)}
+     * returns the first one.
+     * @throws DataStoreException On failure to arrange content into storage
+     */
+    @Test
+    public void nextItemForModelIdReturnsFirstEnqueued() throws DataStoreException {
+        BlogOwner originalJoe = BlogOwner.builder()
+            .name("Joe Swanson")
+            .build();
+        PendingMutation<BlogOwner> firstMutation = PendingMutation.update(originalJoe, BlogOwner.class);
+        storage.save(originalJoe, converter.toRecord(firstMutation));
+
+        BlogOwner updatedJoe = BlogOwner.builder()
```

Comment:
Ah crap this was supposed to say copyOfBuilder, to keep the old ID. Need to validate before push.